pkg/cluster/internal/providers/docker/util.go

```diff
@@ -22,6 +22,16 @@ import (
 	"sigs.k8s.io/kind/pkg/exec"
 )
 
+// IsAvailable checks if docker is available in the system
+func IsAvailable() bool {
+	cmd := exec.Command("docker", "-v")
+	lines, err := exec.OutputLines(cmd)
+	if err != nil || len(lines) != 1 {
+		return false
+	}
+	return strings.HasPrefix(lines[0], "Docker version")
```

Comment:
in the future I think this could probably be relaxed to more like:
- get output (not lines)
- check if output has substring